docs/intro/overview.rst

```diff
@@ -19,143 +19,221 @@ so you can get an idea of how it works and decide if Scrapy is what you need.
+help you extract that information. You need to create Scrapy Spider.
+
+A Spider is just an asynchronous web client surrounded by 
+couple of mechanisms that are processing requests and data returned after
+parsing responses. 
+
+Spider gives you following benefits over plain synchronous HTTP library such 
+as urllib or requests, Scrapy Spider:
+
+* handles typical HTTP scenarios, for example manages cookies,
+  redirects, retries on HTTP error codes, filters duplicate
+  requests, limits spider speed, adds headers (such as 'User-Agent');
+* is asynchronous (non-blocking), it can download and process requests 
+  concurrently. This means that Scrapy Spider will always be faster 
+  and more efficient then synchronous client;
+* makes data extraction easier by allowing you to define common
+  data models;
+* allows you to isolate code that validates and processes data
+  from code that extracts it and deals with response-request cycle, 
+  you can define item pipelines that will process your data, and 
+  middleware that will deal with responses and requests;
+* is fault tolerant, exceptions in one callback processing
+  spider responses do no affect other callbacks.
+
+Let's illustrate these points by building simple
+self-contained spider. Let's say we want to extract the URL, name and 
+number of forks for trending Python repositories in Github. 
+The list of trending repos can be found here: https://github.com/trending?l=python
+
+We can see that this page displays name, url and number of stars for 
+each repo, but it does not show number of forks. To get number of
+forks we will have to make request to each repository page and extract
+data from there. 
 
-Let's say we want to extract the URL, name, description and size of all torrent
-files added today in the `Mininova`_ site.
+.. _intro-overview-item:
 
-The list of all torrents added today can be found on this page:
+We need to write a Spider which visits some initial url
+(in our case https://github.com/trending?l=python), extracts links, 
+follows them by issuing GET requests and extracts data from repository pages.
 
-    http://www.mininova.org/today
+.. highlight:: html
 
-.. _intro-overview-item:
+Extract and follow links
+========================
 
-Define the data you want to scrape
-==================================
+Let's start with link extraction. If we open our browser developer tools 
+and look at page source we'll see that all URLs leading to repositories 
+are wrapped in html that looks like this::
 
-The first thing is to define the data we want to scrape. In Scrapy, this is
-done through :ref:`Scrapy Items <topics-items>` (Torrent files, in this case).
+    <h3 class="repo-list-name">
+        <a href="/honnibal/spaCy">
+        .....
+        </a> 
+    </h3>
 
-This would be our Item::
+.. highlight:: none
 
-    import scrapy
+We'll use `XPath`_ for selecting proper links to repositories. In our case the
```

Comment:
+1 for CSS only for the first exposure: the less things we require the user to know, the quicker she'll get something done and feel badass! :)